src/plugins/mf_priority.cpp

```diff
@@ -777,7 +777,9 @@ static int inactive_cb (flux_plugin_t *p,
             flux_jobtap_raise_exception (p, jobid, "mf_priority",
                                          0, "failed to remove job dependency");
 
-        b->held_jobs.erase (b->held_jobs.begin ());
+        // check that there is in fact a job ID to be removed from the list
+        if (b->held_jobs.begin () != b->held_jobs.end ())
```

Comment:
I'm not an expert on C++, but could `if (!b->held_jobs.empty ())` work here and be more readable? I assume `.begin () != .end ()` is just a way to test for a non empty vector? Apologies if I'm mistaken.

I actually wonder if this was the problem though. I don't see how this check could ever be false since a nonzero size for `b->held_jobs` was checked before entering this block. Do you have any ideas?

Reply:
That's a good point. Before flux-framework/flux-core#4406 landed, valgrind was reporting an `invalid read of size 8` at the line where the ID was erased from the list. So that's what made me think that for some reason the list was empty by the time it got to the `erase ()` call, and an additional check was needed.

I don't know for sure if there was some timing that was happening in the scenario where two jobs under the same user referenced the list of held jobs and were canceled one right after the other, so one job transitioned to `job.state.inactive`, and removed the dependency while the other job was also in `job.state.inactive` but before the `erase ()` call. I don't know if that makes any sense, but that's what I had initially thought was the issue.